awx/sso/backends.py

```diff
@@ -423,3 +468,52 @@ def on_populate_user(sender, **kwargs):
     if profile.ldap_dn != ldap_user.dn:
         profile.ldap_dn = ldap_user.dn
         profile.save()
+
+    reconcile_users_org_team_mappings(user, desired_org_states, desired_team_states, 'LDAP')
```

Comment:
A big non-obvious performance tweak I'd like to have is to handle the case that `desired_org_states` and `desired_team_states` are effectively nothing, in practice, `{}`

That could come in the form of not calling this method when those are empty, or, inside of that method we could bail early. Because the `ContentType.objects.get_for_model` has a query cost and `user_roles` is potentially costly in terms of the volume of return data, and we just don't want to run those when there's nothing to reconcile.

This might already be handled upstream of this, and if it is, you can disregard this comment.